reset.sh

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env bash
+
+# The purpose of this script is to make it easy to reset a local onpremise
+# install to a clean state, optionally targeting a particular version.
+
+set -euo pipefail
+
+if [ -n "${DEBUG:-}" ]; then
+  set -x
+fi
+
+cd "$(dirname $0)"
+
+
+function confirm () {
+  read -p "$1 [y/n] " confirmation
+  if [ "$confirmation" != "y" ]; then
+    echo "Canceled. 😅"
+    exit
+  fi
+}
+
+
+# If we have a version given, validate it.
+# ----------------------------------------
+# Note that arbitrary git refs won't work, because the *_IMAGE variables in
+# .env will almost certainly point to :latest. Tagged releases are generally
+# the only refs where these component versions are pinned, so enforce that
+# we're targeting a valid tag here. Do this early in order to fail fast.
+
+version="${1:-}"
+if [ -n "$version" ]; then
+  set +e
+  git rev-parse --verify --quiet "refs/tags/$version" > /dev/null
```

Comment:
You shouldn't need `> /dev/null` here due to the `--quiet` option.

Love the early verification before doing anything ❤️

Reply:
`--quiet` mutes stderr output, the stdout redirect suppresses the noise of a successful match.

```
$ git rev-parse --verify --quiet refs/tags/21.3.0
0dc51be35274db95c42c00f53d7f56c2aac4d84c
$
```

Reply:
Ugh, you're right sorry. Turns out it only suppresses error messages 🤦🏻